### PR TITLE
Add roles and addon resizer

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/5009

### CODE REVIEW

File: helm/kubernetes-kube-state-metrics-chart/values.yaml [hunk 1]
```diff
@@ -12,15 +12,7 @@ replicas: 1
 image:
   registry: quay.io
   repository: giantswarm/kube-state-metrics
-  tag: v1.4.0
-
-resources:
-  limits:
```
Comment (pipo02mix) [inline helm/kubernetes-kube-state-metrics-chart/values.yaml]: Now they are calculated by the addon resizer

File: helm/kubernetes-kube-state-metrics-chart/templates/rbac.yaml [hunk 2]
```diff
@@ -83,3 +83,37 @@ rules:
   verbs:
   - list
   - watch
+---
```
Comment (pipo02mix) [inline helm/kubernetes-kube-state-metrics-chart/templates/rbac.yaml]: Adding role and role  binding for the addon resizer

File: helm/kubernetes-kube-state-metrics-chart/templates/psp.yaml [hunk 3]
```diff
@@ -6,21 +6,26 @@ metadata:
...
 spec:
-  allowPrivilegeEscalation: true
-  fsGroup:
-    rule: RunAsAny
-  privileged: true
+  privileged: false
   runAsUser:
-    rule: RunAsAny
+    rule: MustRunAsNonRoot
   seLinux:
```
Comment (pipo02mix) [inline helm/kubernetes-kube-state-metrics-chart/templates/psp.yaml]: I have decided to use the same psp configuration as upstream as it way more secure

File: helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml [hunk 4]
```diff
@@ -35,7 +35,34 @@ spec:
             port: {{ .Values.port }}
           initialDelaySeconds: 5
           timeoutSeconds: 5
+      - name: addon-resizer
+        image: k8s.gcr.io/addon-resizer:1.8.3
```
Comment (puja108) [inline helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml]: please retag

**Review by puja108 — COMMENTED** [review]
minor nitpick on retagging, also can you link to upstream files in the PR so it's easier to check? otherwise LGTMIIWFY

File: helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml (hunk 4 above)
Comment (pipo02mix) [inline helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml]:
```suggestion
        image: quay.io/giantswarm/addon-resizer:1.8.3
```

**Comment by pipo02mix** [comment]
@puja108 PTAL retagged

File: helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml [hunk 5]
```diff
@@ -36,7 +36,7 @@ spec:
           initialDelaySeconds: 5
           timeoutSeconds: 5
       - name: addon-resizer
-        image: k8s.gcr.io/addon-resizer:1.8.3
+        image: quay.io/giantswarm/addon-resizer:1.8.3
```
Comment (puja108) [inline helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml]: I think we use the templating here, cause in China we would use the retagged image on alyin, right @rossf7 ?

Comment (pipo02mix) [reply helm/kubernetes-kube-state-metrics-chart/templates/deployment.yaml]: Solved

**Review by puja108 — APPROVED** [review]

**Pull request merged**